JuliaInterface/gap/JuliaInterface.gi

```diff
@@ -57,11 +57,81 @@ InstallMethod( CallFuncList,
     end );
 
 
-##
-##  We support '<juliaobj>[<i>]' and '<juliaobj>[<i>, <j>]' in general.
-##  This is useful for example if <juliaobj> is a tuple.
-##
-BindJuliaFunc( "getindex", "Base" );
+BindGlobal( "_JULIA_MODULE_TYPE", JuliaEvalString( "Module" ) );
+BindGlobal( "_JULIA_FUNCTION_TYPE", JuliaEvalString( "Function" ) );
+BindGlobal( "_JULIA_ISA", JuliaFunction( "isa" ) );
+
+BindGlobal( "_WrapJuliaModule",
+  function( name, julia_pointer )
+    local module;
+
+    module := rec( storage := rec( ),
+                   julia_pointer := julia_pointer );
+
+    ObjectifyWithAttributes( module, TheTypeOfJuliaModules,
+                             Name, Concatenation( "<Julia module ", name, ">" ) );
+
+    return module;
+
+end );
```

Comment:
If you use `ObjectifyWithAttributes` then you can turn the whole function body into a call
`return ObjectifyWithAttributes( rec( ... ), ... );`.

Reply:
Okay, we can keep that in mind, but I think it is more readable this way.